engine/src/main/java/org/destinationsol/game/GalaxyFiller.java

```diff
@@ -290,4 +323,71 @@ public Vector2 getMainStationPosition() {
         return mainStationPos;
     }
 
+    /**
+     * Returns the closest vector to a position from an ArrayList.
+     * @param position The vector that others should be close to.
+     * @param others The array list of other vectors.
+     * @return The nearest vector from others.
+     */
+    private Vector2 getNearestVector(Vector2 position, ArrayList<Vector2> others) {
+        float minDst = Float.MAX_VALUE;
+        Vector2 nearest = null;
+        for (Vector2 current : others) {
+            float dst = position.dst(current);
+            if (dst < minDst) {
+                minDst = dst;
+                nearest = current;
+            }
+        }
+        return nearest;
+    }
+
+    /**
+     * Adjusts positions of the given vector so that it complies with the given constraints.
+     * @param min The minimum distance from the "nearest" vector.
+     * @param max The maximum distance from the "nearest" vector.
+     * @param stationPos The current position that will be changed.
+     * @param nearest The vector that "stationPos" should be min to max distance away.
+     * @return The new position.
+     */
+    private Vector2 computePositionWithConstraints(float min, float max, Vector2 stationPos, Vector2 nearest) {
+        boolean wrongDistance = stationPos.dst(nearest) > max || stationPos.dst(nearest) < min;
+        float modifier = (max-min)/MODIFIER_CONSTANT;
+
+        while (wrongDistance) {
+
+            float gapX = Math.abs(stationPos.x - nearest.x);
+            float gapY = Math.abs(stationPos.y - nearest.y);
+
+            if (gapX * Math.sqrt(2) >= max) {
+                if (stationPos.x > nearest.x) {
+                    stationPos.sub(modifier, 0);
+                } else {
+                    stationPos.add(modifier, 0);
+                }
+            } else if (gapX * Math.sqrt(2) <= min) {
+                if (stationPos.x > nearest.x) {
+                    stationPos.add(modifier, 0);
+                } else {
+                    stationPos.sub(modifier, 0);
+                }
+            } else {
+                if (gapY * Math.sqrt(2) >= max) {
+                    if (stationPos.y > nearest.y) {
+                        stationPos.sub(0, modifier);
+                    } else {
+                        stationPos.add(0, modifier);
+                    }
+                } else if (gapY * Math.sqrt(2) <= min) {
+                    if (stationPos.y > nearest.y) {
+                        stationPos.add(0, modifier);
+                    } else {
+                        stationPos.sub(0, modifier);
+                    }
+                }
+            }
```

Comment:
Hmm, there seems to be duplicate code here. Could you try refactoring it to reduce the duplicate code?